kasparovd/controllers/utxo.go

```diff
@@ -0,0 +1,69 @@
+package controllers
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/kaspanet/kaspad/util/subnetworkid"
+	"github.com/kaspanet/kasparov/apimodels"
+	"github.com/kaspanet/kasparov/dbaccess"
+	"github.com/kaspanet/kasparov/kasparovd/config"
+	"github.com/pkg/errors"
+)
+
+// GetUTXOsByAddressHandler searches for all UTXOs that belong to a certain address.
+func GetUTXOsByAddressHandler(address string) (interface{}, error) {
+	if err := validateAddress(address); err != nil {
+		return nil, err
+	}
+
+	transactionOutputs, err := dbaccess.TransactionOutputsByAddress(dbaccess.NoTx(), address,
+		"Transaction.AcceptingBlock", "Transaction.Subnetwork")
+	if err != nil {
+		return nil, err
+	}
+
+	nonAcceptedTxIds := make([]uint64, len(transactionOutputs))
+	for i, txOut := range transactionOutputs {
+		if txOut.Transaction.AcceptingBlock == nil {
+			nonAcceptedTxIds[i] = txOut.TransactionID
+		}
+	}
+
+	selectedTipBlueScore, err := dbaccess.SelectedTipBlueScore(dbaccess.NoTx())
+	if err != nil {
+		return nil, err
+	}
+	activeNetParams := config.ActiveConfig().NetParams()
+
+	UTXOsResponses := make([]*apimodels.TransactionOutputResponse, len(transactionOutputs))
+	for i, transactionOutput := range transactionOutputs {
+		subnetworkID := &subnetworkid.SubnetworkID{}
+		err := subnetworkid.Decode(subnetworkID, transactionOutput.Transaction.Subnetwork.SubnetworkID)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("Couldn't decode subnetwork id %s", transactionOutput.Transaction.Subnetwork.SubnetworkID))
+		}
+		var acceptingBlockHash *string
+		var acceptingBlockBlueScore *uint64
+		if transactionOutput.Transaction.AcceptingBlock != nil {
+			acceptingBlockHash = &transactionOutput.Transaction.AcceptingBlock.BlockHash
+			acceptingBlockBlueScore = &transactionOutput.Transaction.AcceptingBlock.BlueScore
+		}
+		isCoinbase := subnetworkID.IsEqual(subnetworkid.SubnetworkIDCoinbase)
+		utxoConfirmations := confirmations(acceptingBlockBlueScore, selectedTipBlueScore)
+		isSpendable := utxoConfirmations > 0 && (!isCoinbase || utxoConfirmations >= activeNetParams.BlockCoinbaseMaturity)
```

Comment:
maybe this one is too picky, feel free to ignore. 
I can see this condition is correct, however it is not very readable. Please consider replacing with an if-else statement that will distinguish between isCoinbase or not.

Reply:
I did 
```
isSpendable := (!isCoinbase && utxoConfirmations > 0) ||
			(isCoinbase && utxoConfirmations >= activeNetParams.BlockCoinbaseMaturity)
```
I think I reached a good middle ground.
Let me know if you still don't like it.